api/service_accounts_handlers.go

```diff
@@ -156,12 +157,38 @@ func serviceAccountsListHandler(
 			)
 			return
 		}
-		saSl, count, err := sasUC.WithContext(r.Context()).List(listOptions)
+
+		var saSl []models.ServiceAccount
+		var count int64
+		uc := sasUC.WithContext(ctx)
+		if permissionStr, ok := r.URL.Query()["permission"]; ok {
+			permission, err := models.BuildPermission(permissionStr[0])
+			if err != nil {
+				Write(
+					w, http.StatusUnprocessableEntity,
+					fmt.Sprintf(`{ "error": "%s"  }`, err.Error()),
+				)
+				return
+			}
+
+			// Check if user have that permission
+			saID, _ := getServiceAccountID(ctx)
```

Comment:
You should be aiming to call one usecase function in a handler. So create a new function under usecases.ServiceAccounts that receives formatted input from the handler and returns what's expected to be in the response.